dali/c_api/c_api.cc

```diff
@@ -460,7 +465,7 @@ size_t daliNumTensors(daliPipelineHandle* pipe_handle, int n) {
 
 template <typename T>
 static size_t daliNumElementsHelper(dali::DeviceWorkspace* ws, int n) {
-  return ws->Output<T>(n)._num_elements();
+  return ws->Output<T>(n).shape().num_elements();
```

Comment:
Why not have this forwarded from TV? I think it adds a lot of noise to this PR and will make usage of the new structure more cumbersome.